project/searching.py

```diff
@@ -1,30 +1,43 @@
 # STRETCH: implement Linear Search				
 def linear_search(arr, target):
-  
-  # TO-DO: add missing code
 
-   return -1   # not found
+  for i, e in enumerate(arr):
+    if e == target:
+      return i
+  return -1
 
 
 # STRETCH: write an iterative implementation of Binary Search 
 def binary_search(arr, target):
 
   if len(arr) == 0:
     return -1 # array empty
-    
-  low = 0
   high = len(arr)-1
-
+  middle = int(high/2)
   # TO-DO: add missing code
-
+  for i in range(middle):
+    if target == arr[middle]:
+      return middle
+    elif target < arr[middle]:
+      middle = int(middle/2)
+    else:
+      middle += int(middle/2)
+    
   return -1 # not found
 
 
```

Comment:
binary search looks good , I have never seen (middle/2) before and had to look it up and is great for this function